components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityKeyStoreResolverUtil.java

```diff
@@ -38,12 +38,40 @@ public class IdentityKeyStoreResolverUtil {
      */
     public static String buildTenantKeyStoreName(String tenantDomain) throws IdentityKeyStoreResolverException {
 
+        return buildTenantKeyStoreName(tenantDomain, null);
+    }
+
+    /**
+     * Builds the keystore name for a given tenant domain and context.
+     * The tenant domain is sanitized by replacing dots (.) with hyphens (-) to ensure compatibility
+     * with keystore naming conventions. If a context is provided, it is appended to the sanitized
+     * tenant domain with an underscore (_). The method also appends the standard keystore file
+     * extension as defined in {@link IdentityKeyStoreResolverConstants}.
+     *
+     * @param tenantDomain The domain name of the tenant (e.g., "example.com").
+     * @param context       The optional context to append to the tenant keystore name.
+     * @return A sanitized and formatted keystore name for the tenant.
+     * @throws IdentityKeyStoreResolverException If the tenant domain is null, empty, or invalid.
+     */
+    public static String buildTenantKeyStoreName(String tenantDomain, String context)
+            throws IdentityKeyStoreResolverException {
+
+        // Validate tenantDomain argument
         if (StringUtils.isEmpty(tenantDomain)) {
             throw new IdentityKeyStoreResolverException(
                     ErrorMessages.ERROR_CODE_INVALID_ARGUMENT.getCode(),
                     String.format(ErrorMessages.ERROR_CODE_INVALID_ARGUMENT.getDescription(), "Tenant domain"));
         }
+
+        // Sanitize tenant domain: replace '.' with '-'
         String ksName = tenantDomain.trim().replace(".", "-");
+
+        // Append context if provided
+        if (!StringUtils.isEmpty(context)) {
```

Comment:
```suggestion
        if (StringUtils.isNotEmpty(context)) {
```

Reply:
addressed in https://github.com/wso2/carbon-identity-framework/pull/6189/commits/6ca0b3d01b63c8f3f8ee122988af923b204f342b